f2/src/forms/WhenDidItHappenFormSchema.js

```diff
@@ -1,55 +1,236 @@
-import * as Yup from 'yup'
-import { yupSchema } from '../utils/yupSchema'
-
-const { dateSchema } = yupSchema()
-
-const validationSelection = (selection, validation) => ({
-  is: selection,
-  then: validation,
-  otherwise: Yup.number().typeError(''),
-})
-
-const whenDidItHappenForm = Yup.object().shape({
-  incidentFrequency: Yup.string().required('Please make a selection'),
-  startDay: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.DAY),
-  ),
-  startMonth: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.MONTH),
-  ),
-  startYear: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.YEAR),
-  ),
-  endDay: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.DAY),
-  ),
-  endMonth: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.MONTH),
-  ),
-  endYear: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('moreThanOnce', dateSchema.YEAR),
-  ),
-  happenedOnceDay: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('once', dateSchema.DAY),
-  ),
-  happenedOnceMonth: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('once', dateSchema.MONTH),
-  ),
-  happenedOnceYear: Yup.number().when(
-    'incidentFrequency',
-    validationSelection('once', dateSchema.YEAR),
-  ),
-  description: Yup.string(),
-})
-
-export const whenDidItHappenFormSchema = () => {
-  return whenDidItHappenForm
+import React from 'react'
```

Comment:
Should we rename this file to WhenDidItHappenFormValidator?

Reply:
We could rename. The form schema is defined in this file though. Would it be more appropriate to have two files (schema and validator)?

Reply:
Yes, two files are perfect. Schema is a naming convention resevered for Yup